.github/workflows/pr_test_helm_chart.yml

```
@@ -42,25 +62,31 @@ jobs:
         uses: actions/setup-go@v5
         with:
           go-version: ${{ env.GO_VERSION }}
-          cache: false
-      - name: Unit tests for KOF Operator
-        working-directory: ./kof-operator
-        run: |
-          make test
-      - name: Create Kind Cluster
-        uses: helm/kind-action@v1.8.0
-        with:
-          version: v0.23.0
-          cluster_name: kof-cluster
       - name: Checkout KCM repository
         uses: actions/checkout@v4
         with:
           repository: k0rdent/kcm
           ref: main
           path: kcm-repo
+      - name: Restore KCM CLI Cache
+        id: cache-kcm-cli
+        uses: actions/cache@v3
+        with:
+          path: kcm-repo/bin
+          key: kcm-cli-${{ runner.os }}-${{ github.sha }}
+          restore-keys: |
+            kcm-cli-${{ runner.os }}-
       - name: Install KCM CLI
         run: |
-          make -C kcm-repo cli-install
+          if [ ! -f "kcm-repo/bin/" ]; then
+            make -C kcm-repo cli-install
+          fi
+      - name: Save KCM CLI Cache
+        if: steps.cache-kcm-cli.outputs.cache-hit != 'true'
```

Please delete both `if`-s:
* `make` will find existing targets instantly and will create only missing targets if any.
* "Save KCM CLI Cache" will update the cache to include these missing targets if any.
```suggestion
          make -C kcm-repo cli-install
      - name: Save KCM CLI Cache
```